src/components/DateRangePicker.jsx

```diff
@@ -250,7 +250,9 @@ class DateRangePicker extends React.PureComponent {
     //
     // We handle both situations here by using the ` || ` operator to fallback
     // to *event.target** when **relatedTarget** is not provided.
-    if (this.dayPickerContainer.contains(event.relatedTarget || event.target)) return;
+    const { contains } = this.dayPickerContainer;
+    const { relatedTarget, target } = event;
+    if (contains(relatedTarget) || contains(target)) return;
```

Comment:
if relatedTarget doesn’t exist, we shouldn’t invoke contains on it.

also, i suspect `contains` requires a proper `this` value, which means it can’t be destructured off of the ref.

Reply:
@ljharb if relatedTarget or target doesn’t exist, contains return false

Reply:
I understand, but that doesn't mean we should bother to call the function when we can know it's going to return false.

(also if `contains()` by itself doesn't throw, it probably is doing `document.contains` implicitly, so it's always returning true for every element in the DOM)

Reply:
@ljharb Why you add more checks? Is unnecessary for the functionality

Reply:
Calling into DOM functions is slow; any chance we can avoid doing that is a performance win.

Reply:
@ljharb
const { contains } = this.dayPickerContainer;
It is not a DOM function. Do you know how destructuring works?

Reply:
yes it is, since `this.dayPickerContainer` is an HTML element. You can test it yourself and verify that `contains === HTMLElement.prototype.contains`.

There's no need to be hostile, I can assure you that I know very well how destructuring works.

Reply:
@ljharb I'm sorry, I will fix